artemis-core-client/src/main/java/org/apache/activemq/artemis/api/core/RefCountMessage.java

```diff
@@ -49,6 +172,13 @@ public int getDurableCount() {
       return DURABLE_REF_COUNT_UPDATER.get(this);
    }
 
+   protected void registerDebug() {
+      if (debugStatus == null) {
+         debugStatus = new DebugState(this.toString());
+         ObjectCleaner.register(this, debugStatus);
+      }
+   }
```

Comment:
As this is basically a separated-out part of the constructor, to be used by specific subclasses, it should probably go next to it, and maybe note that.

Reply:
There's no constructor defined on this class. I'm moving it to the top of the class with a note about it.